sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuWindowExpression.scala

```diff
@@ -532,10 +532,13 @@ case class GpuSpecifiedWindowFrame(
     case _ => false
   }
 
-  private def boundarySql(expr: Expression): String = expr match {
-    case e: GpuSpecialFrameBoundary => e.sql
-    case UnaryMinus(n) => n.sql + " PRECEDING"
-    case e: Expression => e.sql + " FOLLOWING"
+  private def boundarySql(expr: Expression): String = {
+    val gpuExp = ShimLoader.getSparkShims.toGpuExpression(expr)
+    gpuExp match {
+      case e: GpuSpecialFrameBoundary => e.sql
+      case GpuUnaryMinus(n) => n.sql + " PRECEDING"
```

Comment:
We could avoid shimming this entirely (at least for now) by avoiding the `unapply` and doing something like this instead:
```suggestion
      case u: UnaryMinus => u.child.sql + " PRECEDING"
```